### ISSUE TITLE
multmatrixtransposed test fails locally with master

### ISSUE THREAD

**Issue by GitPaean** [OPM/opm-simulators#1432]
```
test 14
    Start 14: multmatrixtransposed

14: Test command: arch-linux2-c-debug/bin/mpiexec "-np" "1" "OPM-qt/debug/opm-simulators-build/bin/test_multmatrixtransposed"
14: Test timeout computed to be: 1500
14: Running 1 test case...
14: test_multmatrixtransposed: local/dune2.4.1/release2.4.1_O2_g_mpi/include/dune/common/fvector.hh:124: Dune::FieldVector<K, n>::FieldVector(const std::initializer_list<_Tp>&) [with K = double; int SIZE = 2]: Assertion `l.size() == dimension' failed.
14: unknown location(0): fatal error in "testmultmatrixtrans": signal: SIGABRT (application abort requested)
14: OPM-qt/debug/opm-simulators/tests/test_multmatrixtransposed.cpp(34): last checkpoint
```

The line causes the assertion failure problem is in `fvector.hh:124`

```
122     FieldVector (std::initializer_list<K> const &l)
123     {
124       assert(l.size() == dimension);// Actually, this is not needed any more!
```

Not sure if it happens to other people.

**Comment by GitPaean** [comment]
> I can confirm this. it only happens if `NDEBUG` is not defined which probably explains why jenkins succeeds.

Thanks for the confirmation.  Then it is something needs to be fixed.

**Comment by GitPaean** [comment]
closed with #1432
